src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,406 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure.Functions.Cli.Actions.HostActions;
+using Azure.Functions.Cli.Arm.Models;
+using Azure.Functions.Cli.Helpers;
+using Azure.Functions.Cli.Interfaces;
+using Colors.Net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using static Azure.Functions.Cli.Common.Constants;
+using static Azure.Functions.Cli.Extensions.StringExtensions;
+using static Colors.Net.StringStaticMethods;
+
+namespace Azure.Functions.Cli.Common
+{
+    internal class AuthManager : IAuthManager
+    {
+        public AuthManager() { }
+
+        private AuthSettingsFile MiddlewareAuthSettings;
+
+        private WorkerRuntime _workerRuntime;
+
+        public async Task CreateAADApplication(string accessToken, string AADName, WorkerRuntime workerRuntime, string appName)
+        {
+            if (string.IsNullOrEmpty(AADName))
+            {
+                throw new CliArgumentsException("Must specify name of new Azure Active Directory registration with --appRegistrationName parameter.",
+                    new CliArgument { Name = "appRegistrationName", Description = "Name of new Azure Active Directory registration" });
+            }
+
+            if (CommandChecker.CommandExists("az"))
+            {
+                _workerRuntime = workerRuntime;
+
+                List<string> replyUrls;
```

Comment:
Do you need this to be a `List<string>` or can you instead use `string[]`. `string[]` array is preferred since it is less overhead and it is a fixed size, making it easier to reason about how it is used throughout the code (i.e. nobody is going to add to it or remove from it).

Reply:
It's a List<string> in order to get the serialized array (_string serializedArray = JsonConvert.SerializeObject(replyUrls, Formatting.Indented_) that we use to update /authsettings in the proper format. The other parts of the code could use string[], but not this one. I can either change all of them to string[] and then convert to List in the publish to /authsettings scenario, or eat the overhead elsewhere.